pkg/operator/rbaccontroller/rbaccontroller.go

```diff
@@ -0,0 +1,100 @@
+package rbaccontroller
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listerrbacv1 "k8s.io/client-go/listers/rbac/v1"
+	"k8s.io/klog"
+)
+
+const (
+	// annotation used to indicate if unauthenticated user needs to be
+	// removed from the list of subjects for a cluster role binding
+	openShiftManagedRoleBindingAnnotation = "openshiftapiserver.rbac.authorization.openshift.io/managed"
+)
+
+// RBACController updates a cluster role binding after
+// removing unauthenticated user from the list of subjects
+type RBACController struct {
+	kubeClient               kubernetes.Interface
+	clusterRoleBindingLister listerrbacv1.ClusterRoleBindingLister
+	eventRecorder            events.Recorder
+}
+
+func NewRBACController(kubeClient kubernetes.Interface, informerFactory informers.SharedInformerFactory, recorder events.Recorder) factory.Controller {
+
+	c := &RBACController{
+		kubeClient:               kubeClient,
+		clusterRoleBindingLister: informerFactory.Rbac().V1().ClusterRoleBindings().Lister(),
+		eventRecorder:            recorder,
+	}
+
+	return factory.New().
+		WithInformers(informerFactory.Rbac().V1().ClusterRoleBindings().Informer()).
+		WithSync(c.sync).
+		ToController("RBACController", c.eventRecorder)
+}
+
+func (c *RBACController) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	clusterRoleBindingNames := []string{"discovery", "system:basic-user", "system:discovery", "system:openshift:discovery", "cluster-status-binding"}
```

Comment:
comment what is in this list. Why these?

Reply:
added a comment now